### PR TITLE
Add v4.1 docs

### PR DESCRIPTION
**Pull request by IdanHaim**
Need to create a new branch maybe

### CODE REVIEW

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+RavenDB makes it very easy to be embedded within your application, with RavenDB Embedded package you 
+don't need to do anything only download the package and start your own RavenDB Embedded server.
+{CODE-TABS}
+{CODE-TAB:csharp:Sync embedded_example@Server\Embedded.cs /}
+{CODE-TAB:csharp:Async embedded_async_example@Server\Embedded.cs /}
+{CODE-TABS/}
+
+## Getting Started
+---
+* Create a new project (.NET Core, .Net Framework, whatever).
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: .Net -> .NET
.Net Core -> .NET Standard 2.0+ & .NET Core 2.0+
.NET Framework -> .NET Framework 4.6.1+

'whatever' ? :)

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+* Grab the [pre-release bits from MyGet](https://myget.org/feed/ravendb/package/nuget/RavenDB.embedded).
+Install-Package RavenDB.Embedded -Version 4.1.0 -Source https://www.myget.org/F/ravendb/api/v3/index.json
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Wrap this in code block

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* Start a new Embedded Server
+* Ask the embedded client for a document store, and start working with documents.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Get the new Embedded Document Store, and start working with the database

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Server/Embedded.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            {
+                using (var session = store.OpenSession())
+                {
+                    // Your code here
+                }
+            }
+            #endregion
+
+            #region start_server
+            // Will start RavenDB Server with deafult values
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Server/Embedded.cs]: Start RavenDB Embedded Server with default options

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+### Start The Server
+To start a RavenDB Embedded server you only need to get the Instance of the Embedded Server and then call `StartServer` method.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: RavenDB Embedded Server is available under `EmbeddedServer.Instance`. In order to start it call `StartServer` method.

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+{CODE start_server@Server\Embedded.cs /}
+
+For more control on how to start the server just pass to `StartServer` method a [`ServerOptions`](../../csharp/glossary/server-options) object and that`s it.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Server starting parameters can be controlled by passing to `StartServer` an instance of `ServerOptions`.

Remove ServerOptions from glossary and move here. We want to get rid of the glossary.

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+{CODE start_server_with_options@Server\Embedded.cs /}
+
+{NOTE  Without the `ServerOptions`, RavenDB server will start with a default values on 127.0.0.1:{Random Port}  /}
+
+##### Security
+RavenDB Embedded support running a secured server.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: supports running as a secured server

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Just run `Secured` method in [`ServerOption`](../../csharp/glossary/server-options) object.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: ServerOption -> ServerOptions

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+server certificate. You may supply the certificate password using certPassword.
+
+{CODE security2@Server\Embedded.cs /}
+
+This option is useful when you want to protect your certificate (private key) with other solutions such as "Azure Key Vault", "HashiCorp Vault" or even Hardware-Based Protection. RavenDB will invoke a process you specify, so you can write your own scripts / mini programs and apply whatever logic you need. It creates a clean separation between RavenDB and the secret store in use.
+RavenDB expects to get the raw binary representation (byte array) of the .pfx certificate through the standard output.
+In this options you can control on your client certificate and to use in a different certificate for your client.
+
+### Get Document Store
+After Starting the server you can get the DocumentStore from the Embedded Server and start working with RavenDB.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: DocumentStore -> `DocumentStore`

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+Getting the DocumentStore from The Embedded Server is pretty easy you only need to call `GetDocumentStore` or `GetDocumentStoreAsync` with the name of the database you like to work with. 
+
+{CODE-TABS}
+{CODE-TAB:csharp:Sync get_document_store@Server\Embedded.cs /}
+{CODE-TAB:csharp:Async get_async_document_store@Server\Embedded.cs /}
+{CODE-TABS/}
+
+For more control on the process you can call the methods with [`DatabaseOptions`](../../csharp/glossary/database-options) object.
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Merge the DatabaseOptions into the article, remove them from glossary.

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+{CODE get_server_url_async@Server/Embedded.cs /}
+
+## Acknowledgments
+* You can have only one instance of `EmbeddedServer`.
+* {NOTE: Open RavenDB studio in the browser}
+You can open the studio in the browser with `OpenStudioInBrowser` method
+  {CODE open_in_browser@Server/Embedded.cs /}
+{NOTE /}
+* If you don't have dotnet install you can download it from [here](https://www.microsoft.com/net/download/dotnet-core/runtime-2.1.1)
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Link to https://www.microsoft.com/net/download

We need to extend this description. .NET Core SDKs are required along with the runtime. Write that runtime is set in ServerOptions to the one that we compiled the server with, but can be changed.

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+{CODE-TAB:csharp:Async get_async_document_store_with_database_options@Server\Embedded.cs /}}
+{CODE-TABS/}
+
+### Get Server Url
+For getting the server url you only need to call `GetServerUriAsync` method, this async method will return `Uri` object.
+If `StartServer` is not yet complete this method will wait for the server to start.
+
+{CODE get_server_url_async@Server/Embedded.cs /}
+
+## Acknowledgments
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Remarks

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Server/Embedded.cs [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+            EmbeddedServer.Instance.StartServer();
+            #endregion
+
+            #region start_server_with_options
+            
+            EmbeddedServer.Instance.StartServer(new ServerOptions
+            {
+                DataDirectory = "C:/DataDir",
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Server/Embedded.cs]: Change to `C:\\RavenData`

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown [hunk 14]
```diff
@@ -0,0 +1,75 @@
+﻿# RavenDB Embedded
```
Comment (ppekrol) [inline Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: Change to: `Server : Running an embedded instance`

Comment (ppekrol) [reply Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: We need to link this article to the rest of the documentation.

Add small sub-section in Getting Started about this and in related articles of Client API : What is a Document Store.

Add Related articles to this one too.

File: Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown (hunk 11 above)
Comment (IdanHaim) [reply Documentation/4.1/Raven.Documentation.Pages/server/Embedded.dotnet.markdown]: The user needs to download the Binaries

**Pull request closed without merging**